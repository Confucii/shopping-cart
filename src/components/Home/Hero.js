import city from "./images/city.jpeg";
import "./styles/Hero.css";
import { NavLink } from "react-router-dom";

export default function Hero() {
  return (
    <div className="Hero">
      <div className="hero-holder">
        <div className="hero-text">
          Unleash your potential with our power armor - unyielding protection
          and unstoppable strength is all you need to survive in Night City.
        </div>
        <button className="hero-btn">
          <NavLink className="hero-link" to="/shop">
            Shop Now
          </NavLink>
        </button>
      </div>
      <img className="hero-img" src={city} alt="cyberpunk city" />
    </div>
  );
}
