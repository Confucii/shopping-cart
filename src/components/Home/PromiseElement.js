import "./styles/PromiseElement.css";

export default function PromiseElement({ image, text, alt }) {
  return (
    <div className="PromiseElement">
      <img className="img-Promise-Element" src={image} alt={alt} />
      <p className="text-Promise-Element">{text}</p>
    </div>
  );
}
