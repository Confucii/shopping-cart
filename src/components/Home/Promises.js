import "./styles/Promises.css";
import PromiseElement from "./PromiseElement";
import cat from "./images/cat.svg";
import pistol from "./images/pistol.svg";
import shield from "./images/shield-half-full.svg";

export default function Promises() {
  return (
    <div className="Promises">
      <PromiseElement
        image={pistol}
        alt="Pistol"
        text="To effectively safeguard your life and prove dangerous to everyone else we make our suits powerful. Advanced weapons, enhanced physical strength, and built-in medical treatments make you an ultimate fighting machine."
      />
      <PromiseElement
        image={shield}
        alt="Shield"
        text="
Your safety is of utmost importance to us. Advanced materials provide superior protection and comfort, while built-in shock absorbers and hidden compartments make it perfect for urban warriors."
      />
      <PromiseElement
        image={cat}
        alt="Cat"
        text="Our designs are made to be fashionable. With smooth metallic lines, playful colors, and advanced protection features, this suit is the perfect combination of cute and fierce. Some designs feature feline features."
      />
    </div>
  );
}
