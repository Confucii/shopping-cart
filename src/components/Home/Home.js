import "./styles/Home.css";
import Promises from "./Promises";
import Hero from "./Hero";

export default function Home() {
  return (
    <div className="Home">
      <Hero />
      <Promises />
    </div>
  );
}
