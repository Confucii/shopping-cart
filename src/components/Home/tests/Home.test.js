import React from "react";
import renderer from "react-test-renderer";
import Home from "../Home";
import { BrowserRouter } from "react-router-dom";

describe("Header", () => {
  it("renders correct number", () => {
    const home = renderer
      .create(
        <BrowserRouter>
          <Home />
        </BrowserRouter>
      )
      .toJSON();
    expect(home).toMatchSnapshot();
  });
});
