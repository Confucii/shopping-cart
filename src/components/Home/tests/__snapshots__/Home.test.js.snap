// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correct number 1`] = `
<div
  className="Home"
>
  <div
    className="Hero"
  >
    <div
      className="hero-holder"
    >
      <div
        className="hero-text"
      >
        Unleash your potential with our power armor - unyielding protection and unstoppable strength is all you need to survive in Night City.
      </div>
      <button
        className="hero-btn"
      >
        <a
          className="hero-link"
          href="/shop"
          onClick={[Function]}
        >
          Shop Now
        </a>
      </button>
    </div>
    <img
      alt="cyberpunk city"
      className="hero-img"
      src="city.jpeg"
    />
  </div>
  <div
    className="Promises"
  >
    <div
      className="PromiseElement"
    >
      <img
        alt="Pistol"
        className="img-Promise-Element"
        src="pistol.svg"
      />
      <p
        className="text-Promise-Element"
      >
        To effectively safeguard your life and prove dangerous to everyone else we make our suits powerful. Advanced weapons, enhanced physical strength, and built-in medical treatments make you an ultimate fighting machine.
      </p>
    </div>
    <div
      className="PromiseElement"
    >
      <img
        alt="Shield"
        className="img-Promise-Element"
        src="shield-half-full.svg"
      />
      <p
        className="text-Promise-Element"
      >
        
Your safety is of utmost importance to us. Advanced materials provide superior protection and comfort, while built-in shock absorbers and hidden compartments make it perfect for urban warriors.
      </p>
    </div>
    <div
      className="PromiseElement"
    >
      <img
        alt="Cat"
        className="img-Promise-Element"
        src="cat.svg"
      />
      <p
        className="text-Promise-Element"
      >
        Our designs are made to be fashionable. With smooth metallic lines, playful colors, and advanced protection features, this suit is the perfect combination of cute and fierce. Some designs feature feline features.
      </p>
    </div>
  </div>
</div>
`;
